src/components/tl/ucp/allgather/allgather_linear.c

```diff
@@ -81,7 +81,7 @@ ucc_status_t ucc_tl_ucp_allgather_batched_init(
     task->super.post     = ucc_tl_ucp_allgather_linear_start;
     task->super.progress = ucc_tl_ucp_allgather_linear_progress;
     task->allgather_linear.nreqs =
-        nreqs == UCC_ULUNITS_AUTO ? UCC_TL_TEAM_SIZE(tl_team) - 1 : nreqs;
+        nreqs == 0 ? UCC_TL_TEAM_SIZE(tl_team) - 1 : nreqs;
```

Comment:
where do you handle nreqs=UCC_ULUNITS_AUTO  now?

Reply:
currently here, but we might update later if we find something suitable
```
static unsigned long get_num_reqs(const ucc_tl_ucp_team_t *team)
{
    unsigned long reqs =
        UCC_TL_UCP_TEAM_LIB(team)->cfg.allgather_batched_num_posts;
    ucc_rank_t max_req = UCC_TL_TEAM_SIZE(team) - 1;
    reqs = (reqs > max_req || reqs == UCC_ULUNITS_AUTO) ? max_req : reqs;
    return reqs;
}
```